chpl/chpl-auth/src/main/java/gov/healthit/chpl/auth/manager/impl/UserManagerImpl.java

```diff
@@ -231,37 +259,62 @@ public void updateFailedLoginCount(final UserDTO userToUpdate) throws UserRetrie
     @Override
     @Transactional
     public void updateUserPassword(final String userName, final String password) throws UserRetrievalException {
-
         String encodedPassword = encodePassword(password);
+        loginAdminUser();
         UserDTO userToUpdate = securedUserManager.getBySubjectName(userName);
         securedUserManager.updatePassword(userToUpdate, encodedPassword);
-
+        logoutAdminUser();
     }
 
-    //no auth needed. create a random string and assign it to the user
+    // no auth needed. create a random string and create a new reset token row
+    // for the user
     @Override
     @Transactional
-    public String resetUserPassword(final String username, final String email) throws UserRetrievalException {
+    public UserResetTokenDTO createResetUserPasswordToken(final String username, final String email)
+            throws UserRetrievalException {
         UserDTO foundUser = userDAO.findUserByNameAndEmail(username, email);
         if (foundUser == null) {
             throw new UserRetrievalException("Cannot find user with name " + username + " and email address " + email);
         }
 
-        //create new password
+        // create user password reset token
         char[] buf = new char[GENERATED_PASSWORD_LENGTH];
 
         for (int idx = 0; idx < buf.length; ++idx) {
             buf[idx] = SYMBOLS[random.nextInt(SYMBOLS.length)];
         }
         String password = new String(buf);
 
-        //encode new password
-        String encodedPassword = encodePassword(password);
+        // delete all previous tokens from that user that are in the table
+        userResetTokenDAO.deletePreviousUserTokens(foundUser.getId());
 
-        //update the userDTO with the new password
-        userDAO.updatePassword(foundUser.getSubjectName(), encodedPassword);
+        // create new row in reset token table
+        UserResetTokenDTO userResetToken = userResetTokenDAO.create(password, foundUser.getId());
 
-        return password;
+        return userResetToken;
+    }
+
+    // checks that the token was made in the last 24 hours
+    private boolean isTokenValid(UserResetTokenDTO userResetToken) {
+        Date checkDate = userResetToken.getCreationDate();
+        Instant now = Instant.now();
+        return (!checkDate.toInstant().isBefore(now.minus(HOURS_IN_DAY, ChronoUnit.HOURS)))
```

Comment:
I didn't want the number moved to a static variable. I wanted it moved to a property in the properties file. We don't know what the right duration of the link should be. It could be 30 minutes, or 2 hours, or three days. I'd like the property to be something like "resetLinkExpirationTimeInHours" and use that value here